lib/streamlit/elements/widgets/options_selector/feedback_utils.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Callable, Literal
+
+from streamlit.elements.widgets.options_selector.options_selector_utils import (
+    MultiSelectSerde,
+)
+from streamlit.proto.ButtonGroup_pb2 import ButtonGroup as ButtonGroupProto
+
+_thumb_icons = [":material/thumb_up:", ":material/thumb_down:"]
+_face_icons = [
+    ":material/sentiment_sad:",
+    ":material/sentiment_dissatisfied:",
+    ":material/sentiment_neutral:",
+    ":material/sentiment_satisfied:",
+    ":material/sentiment_very_satisfied:",
+]
+_number_stars = 5
+_star_icon = ":material/star:"
+# we don't have the filled-material icon library as a dependency. Hence, we have it here
+# in base64 format and send it over the wire as an image.
+_selected_star_icon = (
+    "<img src='data:image/svg+xml;base64,"
+    "PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjRweCIgdmlld0JveD0i"
+    "MCAtOTYwIDk2MCA5NjAiIHdpZHRoPSIyNHB4IiBmaWxsPSIjNWY2MzY4Ij48cGF0aCBkPSJNNDgwLTI2OSAz"
+    "MTQtMTY5cS0xMSA3LTIzIDZ0LTIxLThxLTktNy0xNC0xNy41dC0yLTIzLjVsNDQtMTg5LTE0Ny0xMjdxLTEwL"
+    "TktMTIuNS0yMC41VDE0MC01NzFxNC0xMSAxMi0xOHQyMi05bDE5NC0xNyA3NS0xNzhxNS0xMiAxNS41LTE4dDI"
+    "xLjUtNnExMSAwIDIxLjUgNnQxNS41IDE4bDc1IDE3OCAxOTQgMTdxMTQgMiAyMiA5dDEyIDE4cTQgMTEgMS41I"
+    "DIyLjVUODA5LTUyOEw2NjItNDAxbDQ0IDE4OXEzIDEzLTIgMjMuNVQ2OTAtMTcxcS05IDctMjEgOHQtMjMtNkw"
+    "0ODAtMjY5WiIvPjwvc3ZnPg=='/>"
+)
```

Comment:
I have updated it 👍 I guess whether to mark mutable structures such as lists as `Final` or use upper-case is good or bad could be debated (see for example [the Google Java Guideline](https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names)), but if this is what we have done in the codebase I am happy to change it!

Reply:
Yeah, I think we have been treating all module-level variables that don't change during runtime as constants with uppercase writing and `Final` annotation. 

> mark mutable structures such as lists as Final or use upper-case is good or bad could be debated

Marking it with upper-case and final will help us to prevent any undesired mutation. You could also use a tuple instead of a list to make it immutable.